src/analyser.cpp

```diff
@@ -1117,12 +1240,785 @@ void Analyser::AnalyserImpl::analyseEquationAst(const AnalyserEquationAstPtr &as
+
+    if (userUnitsMap) {
+        if (unitsName != "dimensionless") {
+            unitsMap[unitsName] = unitsExponent;
+        }
+    } else {
+        if (isStandardUnitName(unitsName)) {
+            for (const auto &iter : standardUnitsList.at(unitsName)) {
+                if (unitsMap.find(iter.first) == unitsMap.end()) {
+                    unitsMap[iter.first] = 0.0;
+                }
+
+                unitsMap[iter.first] += iter.second * unitsExponent;
+            }
+        } else {
+            UnitsPtr units = model->units(unitsName);
+
+            if (units->isBaseUnit()) {
+                auto iter = unitsMap.find(unitsName);
+
+                if (iter == unitsMap.end()) {
+                    unitsMap[unitsName] = unitsExponent;
+                } else {
+                    unitsMap[iter->first] += unitsExponent;
+                }
+            } else {
+                std::string reference;
+                std::string prefix;
+                double exponent;
+                double multiplier;
+                std::string id;
+
+                for (size_t i = 0; i < units->unitCount(); ++i) {
+                    units->unitAttributes(i, reference, prefix, exponent, multiplier, id);
+
+                    if (isStandardUnitName(reference)) {
+                        for (const auto &iter : standardUnitsList.at(reference)) {
+                            if (unitsMap.find(iter.first) == unitsMap.end()) {
+                                unitsMap[iter.first] = 0.0;
+                            }
+
+                            unitsMap[iter.first] += iter.second * exponent * unitsExponent;
+                        }
+                    } else {
+                        updateUnitsMap(model, reference, unitsMap, userUnitsMap,
+                                       exponent * unitsExponent,
+                                       unitsMultiplier + (std::log10(multiplier) + standardPrefixList.at(prefix)) * unitsExponent);
+                    }
+                }
+            }
+        }
+    }
+}
+
+UnitsMap Analyser::AnalyserImpl::multiplyDivideUnitsMaps(const UnitsMap &firstUnitsMap,
+                                                         const UnitsMap &secondUnitsMap,
+                                                         bool multiply)
+{
+    // Multiply/divide the given units maps together, following a multiplication
+    // (multiply = true) or a division (multiply = false).
+
+    UnitsMap res = firstUnitsMap;
+    double sign = multiply ? 1.0 : -1.0;
+
+    for (const auto &units : secondUnitsMap) {
+        auto it = res.find(units.first);
+
+        if (it == res.end()) {
+            res[units.first] = sign * units.second;
+        } else {
+            it->second += sign * units.second;
+
+            if (areNearlyEqual(it->second, 0.0)) {
+                // The units has now an exponent value of zero, so no need to
+                // track it anymore.
+
```

Comment:
Remove

Reply:
Not sure about this one since the comment is about the whole content of this scope. (Also, we would need to amend most/all of my comments in this file and those regarding the generator.)